src/components/Canvas/Editor.tsx

```diff
@@ -292,30 +294,46 @@ export default class Editor extends EventDispatcher {
     const { top, bottom, right, left } = this.gridLayer.getButtonsDimensions();
     const x = coord.x;
     const y = coord.y;
+    const scaledYHeight = lerpRanges(
+      this.panZoom.scale,
+      // this range is inverted because height has to smaller when zoom out
+      this.maxScale,
+      this.minScale,
+      top.height,
+      top.height * this.extensionAllowanceRatio,
+    );
+    const scaledXWidth = lerpRanges(
+      this.panZoom.scale,
+      // this range is inverted because width has to smaller when zoom out
+      this.maxScale,
+      this.minScale,
+      left.width,
+      left.width * this.extensionAllowanceRatio,
+    );
     if (
       x >= top.x &&
       x <= top.x + top.width &&
-      y >= top.y &&
-      y <= top.y + top.height
+      y >= top.y - scaledYHeight &&
+      y <= top.y + scaledYHeight - scaledYHeight + top.height
```

Comment:
```suggestion
      y <= top.y  + top.height
```
Maybe we don't need to add and subtract the same value